docs/P4c-test-documentation.md

```diff
@@ -0,0 +1,235 @@
+     - **`arch`**: Contains architecture-specific tests, covering various P4 architectures.
+     - **`tools`**: Includes tests for tools provided in the P4C repository, such as `p4test` and `p4c-bm2-ss`.
+     - **`error_samples`**: Contains tests that verify the compiler's behavior when encountering erroneous P4 programs.
+     - **`stf_samples`**: Stores `.stf` (Switch Test Framework) files associated with P4 programs, defining packet inputs/outputs for testing.
+     - **`stress_tests`**: Contains larger or more computationally intensive tests to stress-test the compiler.
+
+   - Make sure the file you are adding is placed in the appropriate directory based on the purpose and backend/tool it is testing.
+   - Ensure the file is named descriptively to reflect its purpose (e.g., `example_positive_test.p4`).
+
+
+2. **File Naming Conventions**:
+
+   - For **BMv2-specific tests** with packets, the source file name must match the pattern `*-bmv2.p4`.
+  - For **v1model architecture** tests: `*-bmv2-v1model.p4`.
+
+  - For **PSA architecture** tests: `*-bmv2-psa.p4`.
+  
+   - Additional file name patterns:
+     - For eBPF-specific tests: `*_ebpf.p4` or `*_ebpf-kernel.p4`.
+     - For graph backend tests: `*graph*.p4`.
+     - For UBPF-specific tests: `*_ubpf.p4`.
+
+     - Other specialized tests under subdirectories such as:
+       - `p4tc_samples/` for TC tests.
+       - `fabric_*/` and `omec/` for Fabric and OMEC tests.
+       - `switch_*/` for specific switch samples.
+
+   These naming conventions ensure the CMake configuration files correctly associate test programs with their respective backends or features.
+
+
+3. **Expected Output Files**:
+
+   - Run the P4 compiler on the source file to verify the output:
+     ```bash
+     ./build/p4test path/to/source_file.p4
+     ```
+
+   - If you need to regenerate expected output files for all test cases, use:
+     ```bash
+     P4TEST_REPLACE=1 make check
+     ```
+   - However, this updates outputs for all tests, so use it with caution.
+
+4. **Updating Expected Outputs**:
+
+   - If a change to the P4 compiler modifies the expected outputs, you can regenerate them by running:
+     ```bash
+     P4TEST_REPLACE=1 ./build/p4test path/to/source_file.p4
+     ```
+   - This will update the expected output file for the specified test.
+   - If you need to regenerate expected outputs for all test cases, use:
+     ```bash
+     P4TEST_REPLACE=1 make check
+     ```
+   - Be cautious when using this command, as it updates outputs for all tests.
+
+### Verifying the Test
+
+- Use the CI system or run the test locally to verify that it passes as expected.
+
+---
+
+## **2. Adding "Negative" Tests**
+
+Negative tests are those where the P4 compiler is expected to catch a syntax or semantic error in the input source file.
+
+### Steps for Adding Negative Tests
+
+1. **Source File Location**:
+
+   - Place the source file in the `p4c/testdata/p4_16_errors` directory.
```

Comment:
That is not the only directory with negative tests.  Please list them all, and provide some guidance for developers to know which one to use for a particular test.